### PR TITLE
CI(pyavd): Ignore Jinja generated instructions from coverage

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Ignore Jinja2 instructions from coverage report.
Leveraging this: https://coverage.readthedocs.io/en/latest/excluding.html#advanced-exclusion
sadly cannot exclude multilines

## Component(s) name

`pyavd`

## Proposed changes

Added list of exceptions from what Jinja code is generated

## How to test

Let pipeline execute

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: python-avd/pyproject.toml [hunk 1]
```diff
@@ -72,3 +72,42 @@ profile = "black"
...
+    "if __name__ == .__main__.:",
+
+    # Don't complain about abstract methods, they aren't run:
+    "@(abc\\.)?abstractmethod",
+
+    # Don't complain about TYPE_CHECKING blocks
+    "if TYPE_CHECKING:",
+
+    # Dont' complain about Jinja2 internalcode
+    # Note that except KeyError is a bid sad..
```
Comment (ClausHolbechArista) [inline python-avd/pyproject.toml]:
```suggestion
    # Note that except KeyError is a bit sad..
```

File: python-avd/pyproject.toml [hunk 2]
```diff
@@ -72,3 +72,42 @@ profile = "black"
...
+    "except KeyError:",
+    "@internalcode",
+    "raise TemplateRuntimeError",
+    "^blocks = {}",
+    "^debug_info =.*",
+]
+
+ignore_errors = true
```
Comment (ClausHolbechArista) [inline python-avd/pyproject.toml]: Why?

File: python-avd/tox.ini [hunk 3]
```diff
@@ -30,17 +30,17 @@ deps =
...
 skip_install = true
 deps =
-    coverage
+    coverage[toml]
 commands =
-    coverage report
+    coverage report --rcfile=pyproject.toml
 
 [coverage:report]
 fail_under = 85
```
Comment (ClausHolbechArista) [inline python-avd/tox.ini]: Please update to the new number. Also this should probably move to the rest of the config settings?

File: python-avd/pyproject.toml (hunk 2 above)
Comment (gmuloc) [reply python-avd/pyproject.toml]: https://coverage.readthedocs.io/en/latest/config.html#report-ignore-errors

removed - I think I have ben carrying this around

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**